errs.go

```diff
@@ -152,13 +152,23 @@ func (e *errorT) Error() string {
 // Format handles the formatting of the error. Using a "+" on the format string
 // specifier will also write the stack trace.
 func (e *errorT) Format(f fmt.State, c rune) {
+	var printSeparator bool
 	for i := len(e.classes) - 1; i >= 0; i-- {
 		name := string(*e.classes[i])
 		if len(name) > 0 {
-			fmt.Fprintf(f, "%s: ", name)
+			if printSeparator {
+				fmt.Fprint(f, ": ")
+			}
+			printSeparator = true
+			fmt.Fprint(f, name)
+		}
+	}
+	if len(e.err.Error()) > 0 {
```

Comment:
Rather than have to format the err twice, can we do something like
```suggestion
	if text := e.err.Error(); len(text) > 0 {
```

and use that in the `Fprintf` below?

Reply:
Done.